ksqldb-tools/src/main/java/io/confluent/ksql/tools/migrations/commands/NewMigrationCommand.java

```diff
@@ -18,20 +18,77 @@
 import com.github.rvesse.airline.annotations.Arguments;
 import com.github.rvesse.airline.annotations.Command;
 import com.github.rvesse.airline.annotations.restrictions.Required;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.FileSystemException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 @Command(
     name = "new",
     description = "Creates a new migrations project, directory structure and config file."
 )
 public class NewMigrationCommand extends BaseCommand {
 
+  private static final Logger LOGGER = LoggerFactory.getLogger(NewMigrationCommand.class);
+
   @Required
   @Arguments(description = "the project path to create the directory", title = "project-path")
   private String projectPath;
 
   @Override
   public void run() {
-    throw new UnsupportedOperationException();
+    final long startTime = System.currentTimeMillis();
+
+    if (tryCreateDirectory(projectPath) &&
+        tryCreateDirectory(projectPath + "/migrations") &&
+        tryCreatePropertiesFile(projectPath + "/ksql-migrations.properties")) {
+      final long endTime = System.currentTimeMillis();
+      LOGGER.info("Migrations project directory created successfully (execution time " +
+          (endTime - startTime)/1000.0 + "s)");
+    } else {
+      System.exit(1);
+    }
   }
 
+  private boolean tryCreateDirectory(final String path) {
+    final File directory = new File(path);
+
+    if (directory.exists() && directory.isDirectory()) {
+      LOGGER.info(path + " already exists. Skipping directory creation.");
+      return true;
+    } else if (directory.exists() && !directory.isDirectory()) {
+      LOGGER.error(path + " already exists as a file. Cannot create directory.");
+      return false;
+    }
+
+    try {
+      LOGGER.info("Creating directory: " + path);
+      Files.createDirectories(Paths.get(path));
+    } catch(FileSystemException e) {
+      LOGGER.error("Permission denied: create directory " + path);
+      return false;
+    } catch (IOException e) {
+      LOGGER.error(String.format("Failed to create directory %s: %s", path, e.getMessage()));
+      return false;
+    }
+    return true;
+  }
+
+  private boolean tryCreatePropertiesFile(final String path) {
+    final File configFile = new File(path);
+    if (configFile.exists()) {
+      LOGGER.info(path + " already exists. Skipping file creation.");
```

Comment:
This log message should at least be a warning. It might make sense to even fail the command if the config file already exists, since it doesn't make sense for the user to run `migrations new` in this case, especially when we introduce a new param to specify the server URL as part of the command. (cc @colinhicks for a minor product question, in case he has input)

Reply:
Shouldn't all these messages go to the command output? If the command fails, it just exits with `System.exit(1)`, and nothing shows the user what was wrong unless they go to the log.

Reply:
changed to warn for now, might rethink this after adding the new param